programs/ido-program/tests/ido-program.ts

```diff
@@ -0,0 +1,466 @@
+    let investorRedeemable: PublicKey;
+
+    let poolNative: PublicKey;
+    let poolFiat: PublicKey;
+
+    let poolSigner: PublicKey;
+
+    let nowBn: anchor.BN;
+    let startIdoTs: anchor.BN;
+    let endIdoTs: anchor.BN;
+    let withDrawFiatTs: anchor.BN;
+
+    const payer = anchor.web3.Keypair.generate();
+    const mintAuthority = anchor.web3.Keypair.generate();
+
+    const project = anchor.web3.Keypair.generate();
+    const investor = anchor.web3.Keypair.generate();
+
+    let pool = anchor.web3.Keypair.generate();
+
+    it("Can initialize the program state", async () => {
+        const transferSig = await provider.connection.requestAirdrop(
+            payer.publicKey,
+            10000000000
+        );
+
+        const latestBlockHash = await provider.connection.getLatestBlockhash();
+
+        await provider.connection.confirmTransaction({
+            blockhash: latestBlockHash.blockhash,
+            lastValidBlockHeight: latestBlockHash.lastValidBlockHeight,
+            signature: transferSig,
+        });
+
+        const tx = new Transaction();
+
+        tx.add(
+            SystemProgram.transfer({
+                fromPubkey: payer.publicKey,
+                toPubkey: mintAuthority.publicKey,
+                lamports: 2000000000,
+            }),
+            SystemProgram.transfer({
+                fromPubkey: payer.publicKey,
+                toPubkey: project.publicKey,
+                lamports: 2000000000,
+            }),
+            SystemProgram.transfer({
+                fromPubkey: payer.publicKey,
+                toPubkey: investor.publicKey,
+                lamports: 2000000000,
+            })
+        );
+
+        await provider.sendAndConfirm(tx, [payer]);
+
+        fiatMint = await createMint(
+            provider.connection,
+            payer,
+            mintAuthority.publicKey,
+            undefined,
+            0,
+            undefined,
+            undefined,
+            TOKEN_PROGRAM_ID
+        );
+
+        nativeMint = await createMint(
+            provider.connection,
+            payer,
+            mintAuthority.publicKey,
+            undefined,
+            0,
+            undefined,
+            undefined,
+            TOKEN_PROGRAM_ID
+        );
+
+        projectFiat = await createAccount(
+            provider.connection,
+            payer,
+            fiatMint,
+            project.publicKey,
+            undefined,
+            undefined,
+            TOKEN_PROGRAM_ID
+        );
+
+        projectNative = await createAccount(
+            provider.connection,
+            payer,
+            nativeMint,
+            project.publicKey,
+            undefined,
+            undefined,
+            TOKEN_PROGRAM_ID
+        );
+
+        investorFiat = await createAccount(
+            provider.connection,
+            payer,
+            fiatMint,
+            investor.publicKey,
+            undefined,
+            undefined,
+            TOKEN_PROGRAM_ID
```

Comment:
These are all optional parameters which doesn't need to be passed since you are using the default token program id anyway.
```suggestion
```